charts/elastic/1.2.2/templates/autoscaling-elastic.yaml

```diff
@@ -0,0 +1,37 @@
+{{- if .Values.elasticsearch.enabled }}
+---
+apiVersion: autoscaling.k8s.elastic.co/v1alpha1
+kind: ElasticsearchAutoscaler
+metadata:
+  name: {{ include "common.names.shortname" $ }}
+  labels:
+  {{- include "common.labels.labels" $ | nindent 4 }}
+  namespace: {{ include "common.names.namespace" $ }}
+spec:
+  {{- with .Values.elasticsearch.params }}
+  ## The name of the Elasticsearch cluster to be scaled automatically.
+  elasticsearchRef:
+    name: {{ include "common.names.shortname" $ }}
+  ## The autoscaling policies.
+  policies:
+  {{- range $k, $v := .roles }}
+    {{- if $v.autoscaling.enabled }}
+    - name: {{ $k }}
+      roles: {{ $v.roles }}
+      resources:
+        nodeCount:
+          min: {{ $v.count }}
+          max: {{ $v.autoscaling.max_nodes }}
+        cpu:
+          min: {{ $v.cpu }}
+          max: {{ $v.autoscaling.max_cpu }}
```

Comment:
Default values defined in values.yaml under:

      data:
        autoscaling:
          enabled: false
          **max_nodes: 4**
          **max_cpu: 4000m**
          max_memory: 8Gi
          max_storage: 200Gi
        count: 3
        **cpu: 4000m**
        memory: 8Gi
        storage: 200Gi

Reply:
You defined default params for data node section. What happen when user accidentally copy master role rather than data role for modification purposes? In other words change master and turn on autoscaling without passing params?  He  gets an error